### DISCUSSION TITLE
x-on:click and @click - Executed Twice

### DISCUSSION THREAD

**Discussion by mendylivium** [livewire/livewire#5987] [1. Help] [answered]
I just observe **x-on:click** and **@click** is executed twice, while **wire:click** works fine on me,

using **x-on:click** and **@click**
![image](https://github.com/livewire/livewire/assets/51048561/c3d8c841-1493-48d2-a78d-e949ff00be95)

using **wire:click**
![image](https://github.com/livewire/livewire/assets/51048561/cf5c5309-3f43-4987-9cd2-6ebc3363969d)

**Comment by N1ebieski** [comment]
Try .prevent > https://livewire.laravel.com/docs/actions#event-handler-modifiers

**Reply by mendylivium** [reply]
I tried, but the result is the same.

Also I try to put console.log('Test') on x-on:click

like these:
```
<button x-bind:disabled="isProcessing" class="btn btn-primary mr-2"
       x-on:click="$dispatch('change-wallet');isProcessing=true;console.log('Test')">Change Wallet
</button>
```
The console.log executed only once. but in the dispatch
![image](https://github.com/livewire/livewire/assets/51048561/a7a516e6-242b-4ccb-9eb0-5e53b94c6fdc)

On the Network Response(Browser)
![image](https://github.com/livewire/livewire/assets/51048561/5700c9e0-1b6b-4fb6-b85f-a2cb2daf5c40)

On the Network Preview(Browser) ther are 2 Payloads
![image](https://github.com/livewire/livewire/assets/51048561/2958c000-b899-4a6a-ac31-976577f26521)

**Comment by mendylivium** [comment, marked as answer]
Problem Solve!

I just change $dispatch() into $wire.dispatch()

Old Code: (Sends Twice the Dispatch)
```
<button x-bind:disabled="isProcessing" class="btn btn-primary mr-2"
       x-on:click="$dispatch('change-wallet');isProcessing=true;console.log('Test')">Change Wallet
</button>
```

New Code:
```
<button x-bind:disabled="isProcessing" class="btn btn-primary mr-2"
       x-on:click="$wire.dispatch('change-wallet');isProcessing=true;console.log('Test')">Change Wallet
</button>
```
